File: Lab9/t3.cpp
#include <stdio.h>
#include <string.h>
int main(){
    char password[20], confirm[20];
    while(1){
        puts("Please set your password: ");
        scanf("%20s", password);
        printf("pass: %s\n", password);
        if (strlen(password) < 8){//Check if the length of password is proper
            puts("Password is not long enough.");
            continue;
        }
        int c;//Variable to judge if we need to flush buffer
        while((c = getchar()) != '\n' && c != EOF);//Flush buffer
        puts("Please confirm your password: ");
        scanf("%20s", confirm);
        printf("confirm: %s\n", confirm);
        if (strcmp(password, confirm) != 0){//Check if the second input is the same
            puts("Passwords don't match.");
            while((c = getchar()) != '\n' && c != EOF);//Flush buffer
            continue;
        }
        printf("Your password \"%s\" is confirmed.\n", confirm);
        break;
    }
    return 0;
}

File: Lab9/t3.java
import java.util.Scanner;
public class t3{
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String password, confirm;
        while(true){
            System.out.println("Please input your password: ");
            password = scan.next();
            if (password.length() < 8){
                System.out.println("Password is not long enough.");
                continue;
            }
            System.out.println("Please confirm your password");
            confirm = scan.next();
            if (!password.equals(confirm)){
                System.out.println("Password don't match.");
                continue;
            }
            System.out.printf("Your password \"%s\" is confirmed.\n", confirm.substring(0, 20));
            break;
        }
    }
}